tools/github_update.py

```diff
@@ -96,118 +105,237 @@ def _get_repo_path(self):
         return re_match.group(1)
 
 
-    def _get_details_link_url(self, details_url):
-        '''returns the url to be included as the details link in the status'''
-        if not details_url:
-            details_url = os.environ.get('GITHUB_COMMIT_STATUS_URL', '') # custom URL via env
-        if not details_url:
-            details_url = os.environ.get('BUILD_URL', '') # provided by jenkins
-            if details_url:
-                details_url += 'console'
-        if not details_url:
-            raise Exception(
-                'Failed to determine URL for details link. ' +
-                'Provide either GITHUB_COMMIT_STATUS_URL or BUILD_URL in env.')
-        return details_url
-
-
-    def _get_auth_token(self):
+    def github_auth_token(self):
         github_token = os.environ.get('GITHUB_TOKEN_REPO_STATUS', '')
         if not github_token:
             github_token = os.environ.get('GITHUB_TOKEN', '')
         if not github_token:
             raise Exception(
                 'Failed to determine auth token to use with GitHub. ' +
                 'Provide either GITHUB_TOKEN or GITHUB_TOKEN_REPO_STATUS in env.')
-        encoded_tok = base64.encodestring(github_token.encode('utf-8'))
-        return encoded_tok.decode('utf-8').rstrip('\n')
+        return github_token
+
 
+class GithubAPI(object):
 
-    def _build_request(self, state, message, details_url = ''):
-        '''returns everything needed for the HTTP request, except the auth token'''
+    def __init__(self, repo_orgname, commit_sha, github_token):
+        self._repo_orgname = repo_orgname
+        self._commit_sha = commit_sha
+        self._github_token = github_token
+
+
+    def _build_post_update_request(self, context_label, state, message, details_url):
```

Comment:
Sorry, I meant it's "buliding commit status" requests, but the method name doesn't say anything about commit statuses.